pynvml/tests/test_smi.py

```diff
@@ -0,0 +1,77 @@
+from pynvml.smi import *
+import pytest
+import os
+
+# Fixture to initialize and finalize nvml
+@pytest.fixture(scope='module')
+def get_instance(request):
+    pytest.nvsmi = nvidia_smi.getInstance()
+
+@pytest.fixture
+def get_count(get_instance):
+    pytest.ngpus = pytest.nvsmi.DeviceQuery('count')['count']
+    print('['+str(pytest.ngpus)+' GPUs]', end =' ')
+    assert(pytest.ngpus>0)
+
+## ---------------------------- ##
+##       Start Query Tests      ##
+## ---------------------------- ##
+
+# Test free-memory query
+def test_query_memory(get_count):
+    def _free_mem_query(j):
+        return pytest.nvsmi.DeviceQuery('memory.free')['gpu'][j]['fb_memory_usage']['free']
+    for i in range(pytest.ngpus):
+        free_mem = _free_mem_query(i)
+        assert(free_mem >= 0)
+
+# Test gpu-utilization query
+def test_gpu_utilization(get_count):
+    def _gpu_util_query(j):
+        return pytest.nvsmi.DeviceQuery('utilization.gpu')['gpu'][j]['utilization']['gpu_util']
+    for i in range(pytest.ngpus):
+        gpu_util = _gpu_util_query(i)
+        assert(gpu_util >= 0)
+
+# Test memory-utilization query
+def test_memory_utilization(get_count):
+    def _memory_util_query(j):
+        return pytest.nvsmi.DeviceQuery('utilization.memory')['gpu'][j]['utilization']['memory_util']
```

Comment:
I'm curious, why separate this line out into a separate function definition?  Why not use it directly in the for loop below?

Reply:
Right - I separated the query into a function only because the query syntax was a bit verbose for some cases.  Beyond making things easier for me to read, there is no good reason to do this.

Reply:
OK, I've submitted https://github.com/rjzamora/pynvml/pull/1 as a proposed alternate.  

I change both how we handle fixtures (which I think we should change to match current idioms) and the pattern used here for tests (which is subjective style, but I thought I'd put it out anyway).

Reply:
We'll want to change the existing tests here to the new style I think.